include/hipSYCL/runtime/data.hpp

```diff
@@ -399,14 +399,13 @@ class data_region
 
     for(std::size_t i = 0; i < 3; ++i){
       assert(page_size[i] > 0);
-      if(num_elements[i] != 1)
-        assert(num_elements[i] % page_size[i] == 0);
+      _num_pages[i] = (num_elements[i] + page_size[i] - 1) / page_size[i];
     }
     
-    _num_pages = num_elements / page_size;
-    for(int i = 0; i < 3; ++i)
-      if(_num_pages[i] == 0)
-        _num_pages[i] = 1;
+    
+    for(int i = 0; i < 3; ++i){
+      assert(_num_pages[i] > 0);
```

Comment:
just move the check into the loop before?

Reply:
Done, good catch :)